ci: Add check-compliance on new commits

myrdyr opened the pull request:
Adds a Jenkinsfile so CI can run here.
Add checkpatch script and gitlint rules.

Signed-off-by: Ulrich Myhre <ulrich.solli.myhre@nordicsemi.no>

NordicBuilder commented:
Found the following issues, please fix and resubmit:

## commit message syntax issues

Config Error: Option extra-path must be either an existing directory or file (current value: 'scripts/gitlint')

NordicBuilder commented:
Found the following issues, please fix and resubmit:

## commit message syntax issues

Config Error: Option extra-path must be either an existing directory or file (current value: 'scripts/gitlint')

NordicBuilder commented:
Found the following issues, please fix and resubmit:

## commit message syntax issues

Config Error: Option extra-path must be either an existing directory or file (current value: 'ci-tools/scripts/gitlint')

nvlsianpu commented:
Need [nrf noup] for PR title and commit message as well.

cc @carlescufi

carlescufi commented:
@myrdyr why do you need `spelling.txt`?

myrdyr commented:
@carlescufi I thought it was required for gitlint, but it was apparantly only used by checkpatch. Removed it as well now, and rebased onto the latest master.
